### PR TITLE
Fix ScalafixImplSuite test and add a test for configuration path

### PR DESCRIPTION
**Pull request by mlachkar**
Right now, even if you specify a config file = None, scalafix still reads the default config path. 
The goal is to document this behavior through a test

### CODE REVIEW

**Comment by mlachkar** [comment]
This test was running only for 2.12. It tests the interface implementations.
Now that we cross-compile example-rule, it's possible to run this test for the 3 scala versions.

File: scalafix-tests/unit/src/test/scala/scalafix/tests/cli/ScalafixImplSuite.scala [hunk 1]
```diff
@@ -97,6 +99,47 @@ class ScalafixImplSuite extends AnyFunSuite with DiffAssertions {
...
+      """.stripMargin,
+        charset
+      )
+    val args = api
+      .newArguments()
+      .withConfig(Optional.empty())
+      .withWorkingDirectory(cwd.toNIO)
+    args.validate()
+    assert(
+      args.rulesThatWillRun().asScala.toList.map(_.toString) == List(
```
Comment (bjaglin) [inline scalafix-tests/unit/src/test/scala/scalafix/tests/cli/ScalafixImplSuite.scala:124]: do you really need to fallback on a `toString` for assertion? (same comment for the second assertion)

File: scalafix-tests/unit/src/test/scala/scalafix/tests/cli/ScalafixImplSuite.scala [hunk 2]
```diff
@@ -97,6 +99,47 @@ class ScalafixImplSuite extends AnyFunSuite with DiffAssertions {
     assert(e.isPresent)
     assert(e.get().getMessage.contains("-Ywarn-unused"))
   }
+  test("configure scalafix path") {
```
Comment (bjaglin) [inline scalafix-tests/unit/src/test/scala/scalafix/tests/cli/ScalafixImplSuite.scala:102]: i don't understand this description - isn't the test more about
```suggestion
  test("rulesThatWillRun") {
```

**Review by bjaglin — APPROVED** [review]
:+1: I have added a few minor suggestions

**Pull request merged**